### PR TITLE
Add isFinal flag for (attached) signature V2 packets

### PR DESCRIPTION
**Pull request by akalin-keybase**

### CODE REVIEW

File: verify_stream.go [hunk 1]
```diff
@@ -83,36 +118,60 @@ func (v *verifyStream) readHeader(versionValidator VersionValidator, msgType Mes
...
 
-func (v *verifyStream) readBlock(p []byte) (int, bool, error) {
-	var block signatureBlock
-	_, err := v.stream.Read(&block)
+func readSignatureBlock(version Version, mps *msgpackStream) (signature, payloadChunk []byte, isFinal bool, seqno packetSeqno, err error) {
+	defer func() {
+		if err == nil {
+			// The header packet picks up the zero seqno,
+			// so subtract 1 to compensate for that.
+			seqno--
```
Comment (akalin-keybase) [inline verify_stream.go]: Out of curiosity, why is this part of the signature format but not the encryption one?

**Comment by akalin-keybase** [comment]
This is basically an adaptation of #33 for signing.

File: verify_stream.go (hunk 1 above)
Comment (akalin-keybase) [inline verify_stream.go]: Ah, looking in the encryption/signcryption code, I see where it's decrementing by 1. I'll see if I can make it consistent.

File: common.go [hunk 2]
```diff
@@ -271,3 +283,43 @@ func SingleVersionValidator(desiredVersion Version) VersionValidator {
...
+		return fmt.Errorf("invalid signature state: version=%s, len(chunk)=%d, isFinal=%t", version, len(chunk), isFinal)
+	}
+
+	switch version.Major {
+	case 1:
+		if (len(chunk) == 0) != isFinal {
+			return makeErr()
+		}
+
+		return nil
```
Comment (oconnor663) [inline common.go]: You could leave these intermediate `return nil` lines out and rely on the one at the end?

Comment (akalin-keybase) [inline common.go]: Done. Also fixed `checkCiphertextState`

**Pull request merged**